core/src/components/content/content.tsx

```diff
@@ -420,6 +424,85 @@ export class Content implements ComponentInterface {
     return promise;
   }
 
+  /**
+   * We need to observe the parent element to detect when
+   * <ion-header> or <ion-footer> elements are added/removed
+   * or resized. This ensures the content offset is recalculated
+   * dynamically.
+   */
+  private connectObservers() {
+    if (!Build.isBrowser) {
+      return;
+    }
+
+    const parent = this.el.parentElement;
+    if (!parent) {
+      return;
+    }
+
+    if ('ResizeObserver' in window) {
+      this.resizeObserver = new ResizeObserver(() => {
+        this.resize();
+      });
+    }
+
+    if ('MutationObserver' in window) {
+      this.mutationObserver = new MutationObserver((mutations) => {
+        let shouldUpdate = false;
+
+        for (const mutation of mutations) {
+          if (mutation.type === 'childList') {
+            mutation.addedNodes.forEach((node: any) => {
+              if (node.tagName === 'ION-HEADER' || node.tagName === 'ION-FOOTER') {
+                shouldUpdate = true;
+              }
+            });
+
+            mutation.removedNodes.forEach((node: any) => {
```

Comment:
I think you can just use `HTMLElement` for this? Shouldn't need to use `any`, preferably